src/schema_tools/core.cljc

```diff
@@ -258,8 +269,12 @@
   [schema]
   (walk/prewalk
     (fn [x]
-      (if (and (map? x) (not (record? x)))
-        (assoc (dissoc x (s/find-extra-keys-schema x)) s/Any s/Any)
+      (if (is-map-not-record? x)
```

Comment:
This could probably be cleaner with `cond` (http://clojuredocs.org/clojure.core/cond), i.e. :
```
(cond
  ((complement is-map-not-record?) x) x
  (has-no-maps? (vals x)) ...
```